app/src/main/java/io/gnosis/safe/ui/transactions/TransactionsFragment.kt

```diff
@@ -0,0 +1,115 @@
+package io.gnosis.safe.ui.transactions
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.google.android.material.tabs.TabLayoutMediator
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.R
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.databinding.FragmentTransactionsBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.SafeOverviewBaseFragment
+import io.gnosis.safe.ui.safe.empty.NoSafeFragment
+import pm.gnosis.svalinn.common.utils.visible
+import javax.inject.Inject
+
+class TransactionsFragment : SafeOverviewBaseFragment<FragmentTransactionsBinding>() {
+
+    @Inject
+    lateinit var viewModel: TransactionsViewModel
+
+    private lateinit var pager: TxPagerAdapter
+
+    override fun inject(component: ViewComponent) {
+        component.inject(this)
+    }
+
+    override fun inflateBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentTransactionsBinding =
+        FragmentTransactionsBinding.inflate(inflater, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            pager = TxPagerAdapter(this@TransactionsFragment)
+            txContent.adapter = pager
+            TabLayoutMediator(txTabBar, txContent, true) { tab, position ->
+                when (TxPagerAdapter.Tabs.values()[position]) {
+                    TxPagerAdapter.Tabs.QUEUE -> {
+                        tab.icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_tx_queue_24dp)
+                        tab.text = getString(R.string.tx_tab_queue)
+                    }
+                    TxPagerAdapter.Tabs.HISTORY -> {
+                        tab.icon = ContextCompat.getDrawable(requireContext(), R.drawable.ic_tx_history_24dp)
+                        tab.text = getString(R.string.tx_tab_history)
+                    }
+                }
+            }.attach()
+        }
+
+        viewModel.state.observe(viewLifecycleOwner, Observer { state ->
+            when (state) {
+                is TransactionsState -> {
+                    val noActiveSafe = state.safe == null
+                    pager.noActiveSafe = noActiveSafe
+                    binding.txTabBar.visible(!noActiveSafe, View.INVISIBLE)
+                    handleActiveSafe(state.safe)
+                }
+            }
+        })
+    }
+
+    override fun handleActiveSafe(safe: Safe?) {
+        navHandler?.setSafeData(safe)
+    }
+
+    override fun screenId(): ScreenId? = null
+}
+
+class TxPagerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
```

Comment:
An idea for a refactor, beyond the scope of this PR: maybe we could abstract this a bit and have super class `PagerAdapterWithDefaultState` (with a better name) and we pass all the enum types, a fragment factory and the default state fragment. We could reuse it in `Settings` and `Assets` too.

Reply:
https://app.zenhub.com/workspaces/safe-multisig---mobile-5c1b98250e13551e8aae81eb/issues/gnosis/safe-android/1136